### DISCUSSION TITLE
Support for arbitrary templates / more documentation on the `template` yaml parameter

### DISCUSSION THREAD

**Discussion by peekxc** [quarto-dev/quarto-cli#6446] [Feature Requests]
### Description

Context: I'm trying to use Quarto + [Eleventy](https://www.11ty.dev/) to generate a custom site that uses [eleventy layouts](https://www.11ty.dev/docs/layouts/). Eleventy is a nice static-site generator in part because it supports virtually any template language (ejs, pug, etc.) 

Quarto seems to have limited support for HTML templates in the form of [custom listings](https://quarto.org/docs/websites/website-listings-custom.html#listing-templates) (why this limited to listings only is unclear). For this particular project, I like the idea of using quarto in the way described by the [docusauraus (or hugo) workflow](https://quarto.org/docs/output-formats/docusaurus#workflow): use quarto to render computational documents to markdown documents, and then use 11ty to generate the site from there.  

So the workflow looks like this, for example: 

```{qmd}
---
title: News.qmd
format: 
  gfm:
    variant: +yaml_metadata_block
template: single_md.pug
---
# News 
...
```

**=>** quarto render **=>** 

```{md}
---
title: News.md
template: single_md.pug
---
# News 
...
```

**=>** eleventy **=>** news.html

However, when I try to use quarto (+[a fix for keep-yaml](https://github.com/quarto-dev/quarto-cli/issues/4210)) to render GFM files, it complains that it could not find the data file corresponding to my template. I don't necessarily want quarto to use that yaml parameter as I want to use that yaml parameter to render the template via eleventy.  

I'm left with the following options: 
- Ask quarto support devs to add an eleventy-like markdown format (as is supported for hugo and docusaurus now)
- Omit the `template` yaml parameters for `*.qmd` documents and post-process the generated `*.md` documents
- Figure out how to get quarto to ignore the yaml template parameter 

I'm actually not sure what path would be the recommended way to go. Any suggestions?

**Comment by mcanouil** [comment]
Would you mind sharing some Git repository or something to work on / help you on?

> Ask quarto support devs to add an eleventy-like markdown format (as is supported for hugo and docusaurus now)

Quarto extension mechanism is here for this, see <https://quarto.org/docs/extensions/>.
